### PR TITLE
Catch Error in notifications and catch socket.error

### PR DESCRIPTION
**Pull request by fernandog**
Second round for https://github.com/pymedusa/Medusa/issues/1980

@labrys

### CODE REVIEW

File: medusa/tv/episode.py [hunk 1]
```diff
@@ -273,7 +274,7 @@ def download_subtitles(self, lang=None):
 
             try:
                 notifiers.notify_subtitle_download(self.pretty_name(), subtitle_list)
-            except RequestException as e:
+            except (RequestException, socket.gaierror) as e:
```
Comment (ratoaq2) [inline medusa/tv/episode.py]: 1. This `try/except` should be removed
2. Add RequestException and gaierror to `exception_handler.py`
3. Delegate to exception_handler.handle(): [here](https://github.com/pymedusa/Medusa/blob/49af1f0a36e3072f0ba468428f07d6c6b72e73fe/medusa/subtitles.py#L1033) and [here](https://github.com/pymedusa/Medusa/blob/49af1f0a36e3072f0ba468428f07d6c6b72e73fe/medusa/subtitles.py#L1038)

That's my opinion

File: medusa/search/core.py [hunk 2]
```diff
@@ -217,20 +217,14 @@ def snatch_episode(result):
             sql_l.append(curEpObj.get_sql())
 
         if curEpObj.status not in Quality.DOWNLOADED:
-            # TODO: Remove this broad catch when all notifiers handle exceptions
```
Comment (fernandog) [inline medusa/search/core.py]: @ratoaq2 removed all

**Comment by fernandog** [comment]
@p0psicles i saw you added some TODO in your get_url PR.

About notifiers, about this PR, we can remove all try:except from notifiers that is related to connectivity

**Comment by fernandog** [comment]
@pymedusa/developers waiting approve on this one

**Pull request merged**